zookeeper-server/src/main/java/org/apache/zookeeper/ZooKeeperMain.java

```diff
@@ -249,6 +261,17 @@ protected void connectToZK(String newHost) throws InterruptedException, IOExcept
             System.setProperty(ZKClientConfig.SECURE_CLIENT, "true");
             System.out.println("Secure connection is enabled");
         }
+        if (cl.getOption("waitforconnection") != null) {
+            connectLatch = new CountDownLatch(1);
+        }
+        int timeout = Integer.parseInt(cl.getOption("timeout"));
+        zk = new ZooKeeperAdmin(host, timeout, new MyWatcher(), readOnly);
```

Comment:
No need to create ZooKeeperAdmin twice.
Also move all these changes after line number 289

Reply:
Removed the statement which was creating Zookeeper admin again